capi/src/lib.rs

```diff
@@ -98,7 +95,7 @@ pub unsafe extern "C" fn dump_jemalloc_pprof(buf_out: *mut *mut u8, n_out: *mut
             return JP_FAILURE;
         }
     };
-    let len: size_t = buf.len().try_into().expect("absurd length");
+    let len: size_t = buf.len();
```

Comment:
I am not sure about getting rid of the `try_into`. I think it depends on `libc::size_t` being the same as `usize`. Is that guaranteed on all platforms?

Reply:
This is something that clippy proposed. The code compiled just fine, but I didn't try it out. If you want I can revert to the old code and tell clippy to not apply the [`useless_conversion`](https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion) linter inside of that function

Reply:
yes, let's revert this change and just ignore the lint here

Reply:
[`usize` is defined to be the same as `uintptr_t`](https://rust-lang.github.io/unsafe-code-guidelines/layout/scalars.html#isize-and-usize) (AKA have the same representation as a pointer), which is _different_ from `size_t`, which is the maximum size of an array. This is not usually an issue, except on some platforms like CHERI which store extra information in the pointer. On those platforms, usize will be 128 bits, while size_t is 64 bit.

Reply:
I've reverted the change, instructed clippy to ignore the warning and added an explanation based on @roblabla's comment.